travel/src/main/java/com/tx/travel/service/ExchangeRateServiceLocalImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.tx.travel.service;
+
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+import com.tx.travel.model.ExchangeRate;
+import com.tx.travel.repository.ExchangeRateRepository;
+import com.tx.travel.service.exception.ExchangeRateAlreadyDefinedException;
+import com.tx.travel.service.exception.ExchangeRateAmountInRsdNotValidException;
+import com.tx.travel.service.exception.ExchangeRateCodeNotValidException;
+import com.tx.travel.service.exception.ExchangeRateNotFoundException;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class ExchangeRateServiceLocalImpl implements ExchangeRateService {
+
+  private final ExchangeRateRepository exchangeRateRepository;
+
+  public ExchangeRateServiceLocalImpl(ExchangeRateRepository exchangeRateRepository) {
+    this.exchangeRateRepository = exchangeRateRepository;
+  }
+
+  @Override
+  public List<ExchangeRate> getAllExchangeRates() {
+
+    return exchangeRateRepository.findAll();
+  }
+
+  @Override
+  public ExchangeRate getExchangeRateById(Long id) throws NoSuchElementException {
+
+    Optional<ExchangeRate> exchangeRateById = exchangeRateRepository.findById(id);
+
+    if (exchangeRateById.isPresent()) return exchangeRateById.get();
+    else throw new NoSuchElementException();
```

Comment:
Maybe you could have a custom exception here

Reply:
I will put ExchangeRateNotFoundException